### PR TITLE
 #164139648 Implement functionality to create articles

### PR DESCRIPTION
**Pull request by codinger41**
#### What does this PR do?
Adds the feature for users to be able to create articles
#### Description of Task to be completed?
Users should be able to create articles and before creating articles, they should be able to preview the articles
In this task, I did the following:
- styled article creation page with preview option
- added action to make a request to create articles

#### How should this be manually tested?
- Start the server and visit `localhost:3000/create-article`
- N/A
#### What are the relevant pivotal tracker stories?
https://www.pivotaltracker.com/story/show/164139648

#### Questions:
N/A

### CODE REVIEW

**Review by Dubby20 — DISMISSED** [review]

**Review by olajide1234 — DISMISSED** [review]

**Review by olajide1234 — DISMISSED** [review]

**Review by Dubby20 — DISMISSED** [review]

**Comment by 9jaswag** [comment]
@leksyib you still haven't changed the PR description

File: package.json [hunk 1]
```diff
@@ -38,7 +40,9 @@
     "axios": "^0.18.0",
     "bootstrap": "^4.3.1",
     "express": "^4.16.4",
+    "form-data": "^2.3.3",
```
Comment (9jaswag) [inline package.json:44]: @leksyib why do we need this?

File: src/__tests__/components/preview.spec.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
+/* eslint-disable react/jsx-no-bind */
+import React from 'react';
+import { mount } from 'enzyme';
+import Preview from '../../components/previewArticle/Preview';
+
+describe('Preview shallow tests', () => {
```
Comment (9jaswag) [inline src/__tests__/components/preview.spec.js]: @leksyib I passed a message about how we should name our describe blocks

File: src/components/previewArticle/Preview.jsx [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  } = props;
+  return (
+    <div>
+      {loading && <Loader />}
+      <div className="form-field">
+        <div className="form text-center">
+          <ArticleTitle title={article.title || ''} />
+          <img
+              alt="image"
+              src={article.image ? URL.createObjectURL(article.image) : 'https://via.placeholder.com/700x400'}
```
Comment (9jaswag) [inline src/components/previewArticle/Preview.jsx]: @leksyib save this to a variable and use it here..to make things cleaner

File: src/views/createArticle/index.jsx [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+
+  function updateInput(e) {
+    if (!(e.target.name === 'image')) {
+      return setArticleData({ ...articleData, [e.target.name]: e.target.value });
+    }
+    return setArticleData({ ...articleData, image: e.target.files[0] });
+  }
+
+  function deleteArticle() {
+    window.location = '/create-article';
```
Comment (9jaswag) [inline src/views/createArticle/index.jsx]: @leksyib why are we doing this?

File: src/views/createArticle/index.jsx [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+  function runValidators(e) {
+    e.preventDefault();
+    const validationErrors = validateArticle(articleData);
+    if (validationErrors.length > 0) {
+      return setState({ ...state, component: 'write', errors: validationErrors });
+    }
+    return completeArticleCreation();
+  }
+
+  function updateInput(e) {
```
Comment (9jaswag) [inline src/views/createArticle/index.jsx]: @leksyib please use `event` instead of `e`

**Review by 9jaswag — CHANGES_REQUESTED** [review]
I also have not seen a screenshare for this

File: package.json (hunk 1 above)
Comment (codinger41) [reply package.json:44]: @9jaswag  The actual FormData API was giving me some issues, also I had to get `boundary` while generating a request header but the FormData API wasn't giving the boundary for the header.

File: src/views/createArticle/index.jsx (hunk 5 above)
Comment (codinger41) [reply src/views/createArticle/index.jsx]: Thanks will do.

File: src/views/createArticle/index.jsx (hunk 4 above)
Comment (codinger41) [reply src/views/createArticle/index.jsx]: We want to clear the state and have a fresh page so I refreshed the page. What do you think will work best here?

Comment (9jaswag) [reply src/views/createArticle/index.jsx]: @leksyib A React application should be a SPA...meaning it should not reload. What does the function do?

File: package.json (hunk 1 above)
Comment (9jaswag) [reply package.json:44]: Also why are we using the `function` keyword?

Comment (codinger41) [reply package.json:44]: I was flagged seriously by hound for using `const` but I think it has been fixed. I'll make changes now.

File: .hound.yml [hunk 6]
```diff
@@ -1,9 +1,5 @@
 eslint:
   enabled: true
   config_file: .eslintrc.json
-  eslint:	
   ignore_file: .eslintignore
-
- sass-lint:
-  enabled: true
-  config_file: .sasslintrc.json 
+
```
Comment (9jaswag) [inline .hound.yml]: @leksyib there's an EOL prob here

**Review by 9jaswag — CHANGES_REQUESTED** [review]

File: .hound.yml (hunk 6 above)
Comment (codinger41) [reply .hound.yml]: @9jaswag  fixed

**Review by 9jaswag — APPROVED** [review]

**Pull request merged**